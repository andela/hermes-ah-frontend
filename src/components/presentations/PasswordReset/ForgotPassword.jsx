import { Form, Button } from 'semantic-ui-react';
import { Redirect } from 'react-router-dom';
import React from 'react';
import FormComponent from '../../shared/Form/Form';
import Loader from '../../shared/Loader/Loader';
import './forgotpassword.scss';

/**
 * @description ForgotPasswordPage view method
 * @returns {HTMLDivElement} profile
 * @param {object} e event object
 */
class ForgotPassword extends FormComponent {
  constructor(props) {
    super(props);
    this.state = {
      data: { email: '' },
    };
  }

  /**
   * @param {object} str event object
   * @returns {object} state
   */

  handelClick = async e => {
    e.preventDefault();
    const { data } = this.state;
    const { forgotPassword: userForgotPassword } = this.props;
    await userForgotPassword(data);
  };

  /**
   * @returns {HTMLElement} div
   */
  render() {
    const { isLoadingReducer, user } = this.props;
    const { loader } = isLoadingReducer;
    if (user) return <Redirect to="/" />;
    return (
      <div className="forgot-password">
        {loader && <Loader />}
        <React.Fragment>
          <div className="form-wrap">
            <div className="form-cont">
              <Form unstackable onSubmit={this.handelClick}>
                <p>FORGOT PASSWORD</p>
                {this.renderInput('email', 'Email', 'email')}
                <Button className="submit" type="submit">
                  Reset my password
                </Button>
              </Form>
            </div>
          </div>
        </React.Fragment>
      </div>
    );
  }
}
export default ForgotPassword;
