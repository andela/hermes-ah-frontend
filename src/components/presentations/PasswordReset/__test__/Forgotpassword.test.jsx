import React from 'react';
import { shallow } from 'enzyme';
import ForgotPassword from '../ForgotPassword';

describe('ForgotPssword component', () => {
  it('should match snapshot', () => {
    const wrap = <ForgotPassword />;
    expect(wrap).toMatchSnapshot();
  });

  it('should change state', () => {
    const wrapper = shallow(
      <ForgotPassword isLoadingReducer={{ loader: true }} />
    );
    expect(wrapper.find('div'));
    expect(wrapper.state('data')).toEqual({
      email: '',
    });
    wrapper.setProps({ isLoading: true });
  });
});
