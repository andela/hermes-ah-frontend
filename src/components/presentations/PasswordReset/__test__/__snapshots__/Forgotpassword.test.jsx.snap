// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPssword component should match snapshot 1`] = `<ForgotPassword />`;
